test/test-pages/gcba/sdk-controller.ts

```diff
@@ -0,0 +1,307 @@
+import { getInputValue, writeToLog } from './utils';
+
+let webrtcSdk: GenesysCloudWebrtSdk;
+let pendingSessions = [];
+
+let conversationUpdatesToRender: {[conversationId: string]: IExtendedMediaSession} = { };
+
+export async function initWebrtcSDK (environmentData) {
+  const jwt = getInputValue('jwt-input');
+  if (!jwt) {
+    window.alert('You need a jwt token');
+    throw new Error('Not Authenticated');
+  }
+
+  const options: ISdkConfig = {
+    jwt,
+    originAppId: 'gcba-demo',
+    originAppName: 'gcba-demo',
+    autoAcceptPendingScreenRecordingRequests: true
+  }
+
+  options.environment = environmentData.uri;
+  options.logLevel = 'info';
+  // for sumo debugging
+  // options.optOutOfTelemetry = true;
+
+  const SDK = GenesysCloudWebrtSdk;
+  (window as any).SDK = SDK;
+
+  webrtcSdk = new SDK(options);
+  (window as any).webrtcSdk = webrtcSdk;
+  (window as any).sdk = webrtcSdk;
+
+  connectEventHandlers();
+  exposeGlobalFunctions();
+
+  await webrtcSdk.initialize()
+  writeToLog(`SDK initialized with ${JSON.stringify(options, null, 2)}`);
+  renderUser(webrtcSdk._personDetails, webrtcSdk._orgDetails);
+}
+
+function connectEventHandlers () {
+  webrtcSdk.on('ready', ready);
+  webrtcSdk.on('pendingSession', pendingSession);
+  webrtcSdk.on('cancelPendingSession', cancelPendingSession);
+  webrtcSdk.on('handledPendingSession', handledPendingSession);
+  webrtcSdk.on('sessionStarted', sessionStarted);
+  webrtcSdk.on('sessionEnded', sessionEnded);
+  webrtcSdk.on('trace', trace);
+  webrtcSdk.on('sdkError', error);
+  webrtcSdk.on('disconnected', disconnected);
+  webrtcSdk.on('connected', connected);
+}
+
+function acceptPendingSession ({ conversationId }) {
+  webrtcSdk.acceptPendingSession({ conversationId });
+}
+
+function rejectPendingSession ({ conversationId }) {
+  webrtcSdk.rejectPendingSession({ conversationId });
+}
+
+function endSession ({ conversationId }) {
+  webrtcSdk.endSession({ conversationId });
+}
+
+function exposeGlobalFunctions () {
+  (window as any).acceptPendingSession = acceptPendingSession;
+  (window as any).rejectPendingSession = rejectPendingSession;
+  (window as any).endSession = endSession;
+}
+
+function _getLogHeader (functionName) {
+  return `${functionName}\n---------------------`;
+}
+
+export function disconnectSdk () {
+  const reallyDisconnect = window.confirm('Are you sure you want to disconnect?');
+  if (!reallyDisconnect) {
+    return;
+  }
+
+  webrtcSdk.disconnect();
+  writeToLog('Disconnected -- Reauthenticate to reconnect');
+}
+
+/* --------------------------- */
+/* SDK EVENT HANDLER FUNCTIONS */
+/* --------------------------- */
+
+function ready () {
+  if (!webrtcSdk._personDetails) {
+    webrtcSdk.startScreenShare();
+  }
+  writeToLog('webrtcSDK ready event emitted');
+}
+
+// pendingSession - {id, address, conversationId, autoAnswer}
+function pendingSession (options) {
+  let output = `${_getLogHeader('pendingSession')}
```

Comment:
Codacy has a fix for the issue: ['output' is never reassigned. Use 'const' instead.](https://app.codacy.com/gh/MyPureCloud/genesys-cloud-webrtc-sdk/pullRequest?prid=9596369)
```suggestion
  const output = `${_getLogHeader('pendingSession')}
```